### PR TITLE
pull-pico-changes: add param to hasFieldInvariantAnnotation

### PR DESCRIPTION
**Pull request by lnsun**
PICO requires a new parameter for hasFieldInvariantAnnotation.

### CODE REVIEW

File: checker/src/main/java/org/checkerframework/checker/initialization/InitializationAnnotatedTypeFactory.java [hunk 1]
```diff
@@ -394,7 +397,7 @@ public void postAsMemberOf(
      * @see #computeFieldAccessType
      * @see #getAnnotatedTypeLhs(Tree)
      */
-    private boolean computingAnnotatedTypeMirrorOfLHS = false;
+    protected boolean computingAnnotatedTypeMirrorOfLHS = false;
```
Comment (xingweitian) [inline checker/src/main/java/org/checkerframework/checker/initialization/InitializationAnnotatedTypeFactory.java:400]: I think this has been added in #142. So no need to add it here.

**Comment by lnsun** [comment]
@xingweitian Thanks for the comment!

File: checker/src/main/java/org/checkerframework/checker/initialization/InitializationAnnotatedTypeFactory.java [hunk 2]
```diff
@@ -196,9 +197,11 @@ protected final boolean hasFieldInvariantAnnotation(VariableTree field) {
      * NullnessAnnotatedTypeFactory#hasFieldInvariantAnnotation(VariableTree)} for an example.
      *
      * @param type of field that might have invariant annotation
+     * @param fieldElement the field element
```
Comment (lnsun) [inline checker/src/main/java/org/checkerframework/checker/initialization/InitializationAnnotatedTypeFactory.java:200]: +doc "can be used to check decl anno"

**Comment by lnsun** [comment]
Move to typetools

**Comment by lnsun** [comment]
Merged to typetools

**Pull request closed without merging**